mlrun/artifacts/base.py

```diff
@@ -1006,13 +1082,13 @@ def upload(self):
                 self._upload_file(src_path)
 
     def _upload_body(self, body, target=None):
-        if calc_hash:
-            self.hash = blob_hash(body)
+        if mlrun.mlconf.should_calculate_artifact_hash():
+            self.hash = calculate_blob_hash(body)
         self.size = len(body)
         store_manager.object(url=target or self.target_path).put(body)
 
     def _upload_file(self, src, target=None):
-        if calc_hash:
+        if mlrun.mlconf.should_calculate_artifact_hash():
             self.hash = calculate_local_file_hash(src)
         self.size = os.stat(src).st_size
         store_manager.object(url=target or self.target_path).upload(src)
```

Comment:
Here the 2 functions exist again, though there isn't any code dup between them. You can still implement here what I suggested above, but here it isn't really necessary